staging/src/k8s.io/component-base/metrics/prometheus/restclient/metrics.go

```diff
@@ -143,6 +155,14 @@ func (l *latencyAdapter) Observe(ctx context.Context, verb string, u url.URL, la
 	l.m.WithContext(ctx).WithLabelValues(verb, u.String()).Observe(latency.Seconds())
 }
 
+type sizeAdapter struct {
+	m *k8smetrics.HistogramVec
+}
+
+func (s *sizeAdapter) Observe(ctx context.Context, verb string, u url.URL, size float64) {
+	s.m.WithContext(ctx).WithLabelValues(verb, u.String()).Observe(size)
```

Comment:
Considering this recent cardinality explosion, we might prevent `u.Path` instead of `u.String()`: https://github.com/kubernetes/kubernetes/pull/106539

Reply:
maybe `r.finalURLTemplate()`?

Reply:
If I am not mistaken, `r.finalURLTemplate()` might also contain the query string, so it won't solve the issue that the query string can take an unbounded number of values. In addition, we don't have access to the request here so we can't really call this method. What we should definitely do though is call `Observe` with `r.finalURLTemplate` and set `u.Path` and `u.Host` in the metric as mentioned by @aojea in https://github.com/kubernetes/kubernetes/pull/106539#discussion_r770328567

Reply:
I would recommend keeping the labels and their values same for both metrics here - `RequestLatency` and `RequestSize`

Reply:
yeah, we have to be consistent there